uiowa/uiowa
# AIS RFI Middleware Updates

Resolves #8213 

# How to test

```
ddev blt ds --site education.uiowa.edu && 
ddev drush @education.local config-split:activate sitenow_webform_ais_rfi -y && 
ddev drush @education.local en dblog && 
ddev drush @education.local uli admin/structure/webform/manage/ces_interest_form/handlers &&
ddev drush @education.local watchdog:tail
```
- The last line of the above output will start outputting the logs in the terminal.
- Click the login link to go to the site. Click "Add handler" and add the "AIS RFI Middleware" handler.
- Confirm that you see the following warning:
```
Warning: The AIS RFI Middleware authentication credentials are missing. Please contact the SiteNow team for assistance.
```
- Finish adding the handler without a UUID. Visit the form and submit it (or use the "Test" tab).
- Check the log output in the terminal and confirm you see the following message:
```
AIS RFI Middleware: Authentication credentials are missing. No data was sent to the middleware.
```
- Add the following to the `docroot/sites/education.uiowa.edu/settings/local.settings.php` and replacing the `username` and `password` with the ones stored in esaccess.
```php
$config['sitenow_webform_ais_rfi.settings']['middleware']['auth']['user'] = 'username';
$config['sitenow_webform_ais_rfi.settings']['middleware']['auth']['pass'] = 'password';
```
- Edit the handler (don't change anything though) and confirm the warning on the form is no longer showing.
- Submit the form again and confirm that you see the following message in the logs:
```
AIS RFI Middleware: Interaction UUID is missing. No data was sent to the middleware.
```
- Edit the handler and add the interaction UUID (there is an email in the web team inbox that contains the UUID or you can ping @pyrello for it). Submit the form again and confirm you see the following message:
```
AIS RFI Middleware: Success: Data received and enqueued for processing.
```
